### ISSUE TITLE
Add ec2_tag resource

### ISSUE THREAD

**Issue by mhaley-miovision** [terraform-providers/terraform-provider-aws#11634]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS has a tagging API for all EC2 resources.  Terraform should have a resource of its own to implement this API.

In my particular use case I am looking to tag the ENI created by an AWS NAT gateway.  But I can see more cases such as Lambdas or ELBs where AWS creates the network interfaces but you still want to adhere to your company's tagging strategy.

### New or Affected Resource(s)

* aws_ec2_tags

### Potential Terraform Configuration

```hcl
resource "aws_nat_gateway" "gw" {
  allocation_id = "${aws_eip.nat.id}"
  subnet_id     = "${aws_subnet.public.id}"

  tags = {
    Name = "gw NAT"
  }
}

resource "aws_ec2_tags" "nat_gw_eni" {
  resource_id = aws_nat_gateway.gw.network_interface_id
}
```

### References

* https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateTags.html
* https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeTags.html
* https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DeleteTags.html

**Comment by ewbankkit** [comment]
Similar:
* https://github.com/terraform-providers/terraform-provider-aws/issues/9061

**Comment by scjody** [comment]
It looks like https://github.com/terraform-providers/terraform-provider-aws/pull/8457 addresses this need.

**Comment by mhaley-miovision** [comment]
@scjody it does.  I would suggest to have the ec2_tag resource accept a map of tags instead of needing to specify individual key-value pairs in a resource.

**Comment by bflad** [comment]
A new `aws_ec2_tag` resource for managing individual EC2 resource tags has been merged and will release with version 2.67.0 of the Terraform AWS Provider, later next week. This resource should only be used in cases where EC2 resources are created outside Terraform (e.g. AMIs), being shared via Resource Access Manager (RAM), or implicitly created by other means (e.g. Transit Gateway VPN Attachments).

```hcl
# Example configuration in Terraform 0.12 and later syntax
resource "aws_ec2_transit_gateway" "example" {}

resource "aws_customer_gateway" "example" {
  bgp_asn    = 65000
  ip_address = "172.0.0.1"
  type       = "ipsec.1"
}

resource "aws_vpn_connection" "example" {
  customer_gateway_id = aws_customer_gateway.example.id
  transit_gateway_id  = aws_ec2_transit_gateway.example.id
  type                = aws_customer_gateway.example.type
}

resource "aws_ec2_tag" "example" {
  resource_id = aws_vpn_connection.example.transit_gateway_attachment_id
  key         = "Name"
  value       = "Hello World"
}
```

As with any Terraform 0.12.6 or later configuration, this resource can be combined with [`for_each` support](https://www.terraform.io/docs/configuration/resources.html#for_each-multiple-resource-instances-defined-by-a-map-or-set-of-strings) to manage multiple resource tags, if necessary.

Thanks to @joestump and others who made the implementation possible. 👍
